src/io/opendss.jl

```diff
@@ -968,22 +1418,36 @@ function parse_opendss(dss_data::Dict; import_all::Bool=false, vmin::Float64=0.9
     dss2tppm_shunt!(tppm_data, dss_data, import_all)
     dss2tppm_branch!(tppm_data, dss_data, import_all)
     dss2tppm_transformer!(tppm_data, dss_data, import_all)
+    dss2tppm_reactor!(tppm_data, dss_data, import_all)
     dss2tppm_gen!(tppm_data, dss_data, import_all)
     dss2tppm_pvsystem!(tppm_data, dss_data, import_all)
     dss2tppm_storage!(tppm_data, dss_data, import_all)
 
-    adjust_sourcegen_bounds!(tppm_data)
-
     tppm_data["dcline"] = []
 
     InfrastructureModels.arrays_to_dicts!(tppm_data)
+    # empty branch arrays are not converted into dicts;
+    # resolve this here to prevent issues later on
+    for key in ["branch"]
+        if isa(tppm_data[key], Array)
+            tppm_data[key] = Dict{String, Any}()
+        end
+    end
 
     for optional in ["dcline", "load", "shunt", "storage", "pvsystem"]
         if length(tppm_data[optional]) == 0
             tppm_data[optional] = Dict{String,Any}()
         end
     end
 
+    if haskey(tppm_data, "trans_comp")
+        # this has to be done before calling adjust_sourcegen_bounds!
+        decompose_transformers!(tppm_data)
+        adjust_base!(tppm_data)
+    end
```

Comment:
It looks like `trans_comp` is left in place, but is it used later or only during this parsing process? If it is not used I think after the parsing it should be deleted.

Reply:
After re-reading the notes you sent by email, I think `trans_comp` should be retained only if `import_all` is enabled.

Reply:
One potential use for the trans_comp data would be to create a composed transformer in the solution dictionary.  There the solutions could be shown for the entire transformer, and not dispersed over several other dicts. If there is no reference to the compound elements in the data model,  such a transformation would not make any sense. This feature is low priority for us, but we would like to keep the trans_comp dict around for the future.

I implemented it as you proposed; if import_all is not flagged, trans_comp is removed after the call to decompose_transformers.